Download fails and crashes ckan with FATAL UNHANDLED EXCEPTION (Mono.Unix.UnixIOException: Resource temporarily unavailable [EWOULDBLOCK])

Crash occurs while trying to update "Restock" mod.  Other mods were able to be updated.

ckan v1.30.4; Debian GNU/Linux 11

Have now tried several times, including moving the download cache from my home directory (local disk) to a network share.  Every time close to the end of the download I get this exception. 

See attached exception log (please note that the first exception in this log is immaterial; crash also happens when not attempting to update DistantObject mod).

[ckan_crash.txt](https://github.com/KSP-CKAN/CKAN/files/8424009/ckan_crash.txt)

```
[ERROR] FATAL UNHANDLED EXCEPTION: Mono.Unix.UnixIOException: Resource temporarily unavailable [EWOULDBLOCK].
  at Mono.Unix.UnixMarshal.ThrowExceptionForLastError () [0x00005] in <b239aefa15154eb595f94b279272ec9d>:0 
  at Mono.Unix.UnixStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 count) [0x00055] in <b239aefa15154eb595f94b279272ec9d>:0 
  at System.Windows.Forms.XplatUIX11.WakeupMain () [0x00000] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
  at System.Windows.Forms.XplatUIX11.SendAsyncMethod (System.Windows.Forms.AsyncMethodData method) [0x00080] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
  at System.Windows.Forms.XplatUI.SendAsyncMethod (System.Windows.Forms.AsyncMethodData data) [0x00000] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
  at System.Windows.Forms.Control.BeginInvokeInternal (System.Delegate method, System.Object[] args, System.Windows.Forms.Control control) [0x0003f] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
  at System.Windows.Forms.Control.Invoke (System.Delegate method, System.Object[] args) [0x00017] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate,object[])
  at System.Windows.Forms.Control.Invoke (System.Delegate method) [0x0001d] in <6d635ac3dc1c4424ad385ded79f1e868>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate)
  at CKAN.Util.Invoke[T] (T obj, System.Action action) [0x0000d] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.Wait.SetDescription (System.String message) [0x00014] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at (wrapper remoting-invoke-with-check) CKAN.Wait.SetDescription(string)
  at CKAN.GUIUser.RaiseProgress (System.String message, System.Int32 percent) [0x00017] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.NetAsyncDownloader.FileProgressReport (System.Int32 index, System.Int32 percent, System.Int64 bytesDownloaded, System.Int64 bytesToDownload) [0x0019e] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.NetAsyncDownloader+<>c__DisplayClass16_0.<DownloadModule>b__0 (System.Object sender, System.Net.DownloadProgressChangedEventArgs args) [0x0001e] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at CKAN.NetAsyncDownloader+NetAsyncDownloaderDownloadPart.<ResetAgent>b__20_0 (System.Object sender, System.Net.DownloadProgressChangedEventArgs args) [0x00008] in <933dc9d8b62a44ff9a8f866e3767654d>:0 
  at System.Net.WebClient.OnDownloadProgressChanged (System.Net.DownloadProgressChangedEventArgs e) [0x0000a] in <9c6e2cb7ddd8473fa420642ddcf7ce48>:0 
  at System.Net.WebClient.<StartAsyncOperation>b__78_9 (System.Object arg) [0x00000] in <9c6e2cb7ddd8473fa420642ddcf7ce48>:0 
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x00007] in <533173d24dae460899d2b10975534bb0>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <533173d24dae460899d2b10975534bb0>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <533173d24dae460899d2b10975534bb0>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in <533173d24dae460899d2b10975534bb0>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <533173d24dae460899d2b10975534bb0>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <533173d24dae460899d2b10975534bb0>:0 
```

--- a/Core/Net/NetAsyncDownloader.cs
+++ b/Core/Net/NetAsyncDownloader.cs
@@ -96,19 +96,6 @@ namespace CKAN
             }
         }
 
-        /// <summary>
-        /// Downloads from hosts in this list will be done sequentially rather
-        /// than in parallel
-        /// </summary>
-        private static readonly HashSet<string> throttledHosts = new HashSet<string>()
-        {
-            /// GitHub returns a 403-Forbidden status sometimes if you try to download
-            /// too much in parallel, see https://github.com/KSP-CKAN/CKAN/issues/2210
-            "github.com",
-            "api.github.com",
-            "raw.githubusercontent.com",
-        };
-
         private static readonly ILog log = LogManager.GetLogger(typeof (NetAsyncDownloader));
 
         public readonly IUser User;
@@ -189,8 +176,8 @@ namespace CKAN
 
         /// <summary>
         /// Check whether a given download should be deferred to be started later.
-        /// Decision is made based on whether the host is throttled and whether
-        /// we're already downloading something else from it.
+        /// Decision is made based on whether we're already downloading something
+        /// else from the same host.
         /// </summary>
         /// <param name="target">Info about a requested download</param>
         /// <returns>
@@ -198,8 +185,7 @@ namespace CKAN
         /// </returns>
         private bool shouldQueue(Net.DownloadTarget target)
         {
-            return throttledHosts.Contains(target.url.Host)
-                && downloads.Any(dl =>
+            return downloads.Any(dl =>
                     dl.target.url.Host == target.url.Host
                     && dl.bytesLeft > 0);
         }
@@ -423,16 +409,13 @@ namespace CKAN
                 log.InfoFormat("Finished downloading {0}", downloads[index].target.url);
             }
 
-            if (throttledHosts.Contains(downloads[index].target.url.Host))
+            var next = queuedDownloads.FirstOrDefault(dl =>
+                dl.url.Host == downloads[index].target.url.Host);
+            if (next != null)
             {
-                var next = queuedDownloads.FirstOrDefault(dl =>
-                    dl.url.Host == downloads[index].target.url.Host);
-                if (next != null)
-                {
-                    // Start this host's next queued download
-                    queuedDownloads.Remove(next);
-                    DownloadModule(next);
-                }
+                // Start this host's next queued download
+                queuedDownloads.Remove(next);
+                DownloadModule(next);
             }
 
             onOneCompleted.Invoke(downloads[index].target.url, downloads[index].path, downloads[index].error);

